frontend/workflows/projectSelector/src/project-selector.tsx

```diff
@@ -61,8 +63,12 @@ const StyledProgressContainer = styled.div({
 // Determines if every project has projectData (i.e. the effect has finished fetching the data)
 const allPresent = (state: State): boolean => {
   let ret = true;
-  // We could potentially check all groups but it's really not necessary since an upstream cannot be added manually.
-  _.forEach(Object.keys(state[Group.PROJECTS]), p => {
+  const allProjects = [
```

Comment:
nit: a `Set` would be more appropriate here in order to dedupe. not sure if there's a perf implication either way.

Reply:
Makes sense for uniqueness but imo it wasn't a huge concern since it'll just be an extra validation. Performance wise, at the scale we are realistically rendering, neither make a difference given the use case. I'll udpate.